core/src/main/java/org/kiwix/kiwixmobile/core/history/HistoryActivity.java

```diff
@@ -203,8 +203,13 @@ public boolean onOptionsItemSelected(MenuItem item) {
       onBackPressed();
       return true;
     } else if (itemId == R.id.menu_history_toggle) {
-      item.setChecked(!item.isChecked());
-      sharedPreferenceUtil.setShowHistoryCurrentBook(item.isChecked());
+      if (!sharedPreferenceUtil.getShowHistoryCurrentBook()) {
```

Comment:
the deleted code was more concise, I would revert to it and use the toast extension for showing a toast, using a ternary to choose the stringId eg
```
ContextExtensionsKt.toast(
this, 
item.getChecked() ? R.string.history_current_book : R.string.history_all_books,
 Toast.LENGTH_LONG);
```

That is if we keep this implementation, I am not wild on the triangle/toasts as it feels quite confusing. I'd prefer if we just showed this in the ui with a checkbox/toggle, we can keep it in or below the toolbar

Reply:
@macgills I made the changes according to the guidelines I saw in the issue. 
I also feel that keeping the History option should be with a checkbox. So maybe we could keep the trash icon outside the menu to make the menu more compact and leave the History option in the menu itself?
Please let me know your thoughts on this.

Reply:
I think we don't need to use the `menu` to achieve this. This actionBar is infact a toolbar and you can put arbitrary views in a toolbar. I am picturing it below the menuItems for now.

The current text is a bit annoying `History From Current Book`, by default we only show the current book's history. I wouldn't be surprised if the vast majority of our users don't know this option to see all history exists and I think that better explains what it is.
"View All History" or the more verbose "View All Books History" I think would be good options for the name of this checkbox/toggle. It is difficult to convey what it does concisely so I am open to suggestions. I am personally leaning more towards toggle but follow your heart.

Trash/Search icon should definitely remain in the menu

Reply:
I will update the pr with a better UI by tomorrow.
Thanks a lot for the suggestions :)

Reply:
@macgills I am changing the title of the option so that it's described better. I think the existing checkbox is doing a good job as we intend on keeping this item in the menu itself. 
Regarding the clear history option, I'm making it in such a way that it will be visible only when historyList is not empty. If the historyList is empty, the option to clear history will also be made invisible.

Reply:
I have concern about it still being in the menu because of how hidden it is in the overflow, the sheer size of the text+checkbox make it difficult to fit in the actionBar with any extra icons present so constantly expanded also doesn't seem great. That is what lead me to placing in the toolbar, if that poses a challenge then above the list would also be fine I think

Reply:
@macgills I am not very clear as to how to put the option in the toolbar. I am not able to figure out how to place this option under the menu items.

Reply:
![image](https://user-images.githubusercontent.com/3358282/73275329-dfb5cc00-41de-11ea-92ec-d637b6393676.png)
 place your cursor on `<include layout="@layout/layout_standard_app_bar" />`
and press `ctrl+alt+n`, select `inline this usage`

This will give you the appbar that contains the toolbar, we don't actually need to put the view in the toolbar, apologies for misleading you.

Obviously there are some styling issues with this solution, it looks like [this answer](https://stackoverflow.com/a/58749611/12225251) is a good starting point for making the background themed dark or seeing as how this is the first time we are styling an AppBar I'll allow in file styling

Reply:
Thanks for help @macgills :) 
I'll just update the pr as per these guidelines.

Reply:
I'll make the same changes in the bookmarks section too since it the same feature.